### PR TITLE
Restore jsr-305 dependency

### PR DESCRIPTION
**Pull request by timja**
Untested but I expect it to fix https://github.com/jenkinsci/configuration-as-code-plugin/issues/2491

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
I've confirmed this works with interactive testing in my lts-with-plugins container image.

Tests that I ran include:

* Start Jenkins without it and confirm the message is displayed in the log
* Explore Jenkins to see if there were obvious failures that a user would see (none detected)
* Install into the lts-with-plugins plugins.txt file
* Start Jenkins and confirm the message is not displayed
* Display the configuration as code
* Reload the configuration as code

File: plugin/pom.xml [hunk 1]
```diff
@@ -65,6 +65,11 @@
   </dependencyManagement>
 
   <dependencies>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
```
Comment (MarkEWaite) [inline plugin/pom.xml:69]: As a reminder to others who might try to remove this from the pom.xml file.
```suggestion
            <!-- Do not remove without interactive testing to confirm it is not needed -->
            <!-- https://github.com/jenkinsci/configuration-as-code-plugin/pull/2490#issuecomment-2115366886  -->
            <groupId>com.google.code.findbugs</groupId>
```

Comment (MarkEWaite) [inline plugin/pom.xml:69]:
```suggestion
      <!-- Do not remove without interactive testing to confirm it is not needed -->
      <!-- https://github.com/jenkinsci/configuration-as-code-plugin/pull/2490#issuecomment-2115366886  -->
      <groupId>com.google.code.findbugs</groupId>
```

**Comment by MarkEWaite** [comment]
I was trying to add my proposed extra comment but I think it is wrong to delay the delivery of this fix.  I'll add that comment later when I've investigated further.

**Comment by MarkEWaite** [comment]
Notes that I put into my test commit included:

## Fix init warning message by restoring JSR-305 dependency

https://github.com/jenkinsci/configuration-as-code-plugin/pull/2490 removed the JSR-305 dependency because the code compiles and passes tests without that dependency.  However, a new startup failure message is reported without that dependency.  Retain the dependency for now until more investigation can identify the other changes needed to remove that dependency.

Thanks to @kutzi for reporting it promptly!

https://github.com/jenkinsci/configuration-as-code-plugin/pull/2490#issuecomment-2115366886 is the report.

Fixes #2491

**Comment by basil** [comment]
History repeats itself one year later, almost to the day: #2238

**Comment by MarkEWaite** [comment]
> History repeats itself one year later, almost to the day: #2238

Yes, my mistake that I didn't perform the basic interactive test of loading it into my own configuration and watching the log file.

I'll submit a pull request to add a comment that warns others away from making the same mistake that I made.

**Pull request merged**